include/logconfig.h

```diff
@@ -1,10 +1,13 @@
 #ifndef __LOGCONFIG_H__
 #define __LOGCONFIG_H__
 
+#include <regex>
+
 namespace logger {
 constexpr int THRESHOLD = 0;
 constexpr char filename[] = "hey.txt";
 constexpr char format[] = "%(%:%):";
+std::regex format_regex = std::regex("(.*)\\((.*):(.*)\\):(.*)");
```

Comment:
Was wondering if there is a way to only use regex and infer the `format` field from it? Maybe in `v1.2` we should do it otherwise there might be inconsistency issues if the user doesn't configure it correctly